selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitPullConflictTest.java

```diff
@@ -0,0 +1,199 @@
+  @Inject private TestUserPreferencesServiceClient testUserPreferencesServiceClient;
+  @Inject private TestGitHubServiceClient gitHubClientService;
+
+  @BeforeClass
+  public void prepare() throws Exception {
+    gitHub = GitHub.connectUsingPassword(gitHubUsername, gitHubPassword);
+    gitHubRepository = gitHub.createRepository(REPO_NAME).create();
+    String commitMess = String.format("add-new-content %s ", System.currentTimeMillis());
+    testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
+    Path entryPath = Paths.get(getClass().getResource("/projects/git-pull-conflict").getPath());
+    gitHubClientService.addContentToRepository(entryPath, commitMess, gitHubRepository);
+    ide.open(ws);
+  }
+
+  @AfterClass
+  public void deleteRepo() throws IOException {
+    gitHubRepository.delete();
+  }
+
+  @Test
+  public void pullConflictsTest() throws IOException {
+    // preconditions and import the test repo
+    String javaFileChange = "JavaCommentsTest";
+    String textFileChange = "GitPullTest.txt";
+    String changeContent_2 = "//second_change";
+
+    projectExplorer.waitProjectExplorer();
+    String repoUrl = String.format("https://github.com/%s/%s.git", gitHubUsername, REPO_NAME);
+    git.importJavaApp(repoUrl, PROJECT_NAME, MAVEN);
+
+    // change files in the test repo on GitHub
+    changeContentOnGithubSide(
+        String.format("%s/%s.java", pathToJavaFile, javaFileChange), CHANGE_STRING_1);
+    changeContentOnGithubSide(textFileChange, CHANGE_STRING_1);
+
+    // change the same files in the editor
+    changeJavaFileForTest(javaFileChange, changeContent_2);
+    changeTextFileForTest(textFileChange, changeContent_2);
+
+    // make pull and get the first conflict
+    performPull();
+    events.clickEventLogBtn();
+    events.waitExpectedMessage(firstMergeConflictMessage);
+
+    commitFiles();
+
+    // Make pull again and get second conflict
+    performPull();
+    events.clickEventLogBtn();
+    events.waitExpectedMessage(secondMergeConflictMessage);
+
+    // Checking the message has present
+    editor.selectTabByName(javaFileChange);
+    editor.waitActive();
+    editor.waitTextIntoEditor(headConfPrefixConfMess);
+    editor.selectTabByName(textFileChange);
+    editor.waitActive();
+    editor.waitTextIntoEditor(headConfPrefixConfMess);
+  }
+
+  private void changeJavaFileForTest(String fileName, String text) {
+    projectExplorer.quickExpandWithJavaScript();
+    projectExplorer.openItemByPath(
+        String.format("%s/%s/%s.java", PROJECT_NAME, pathToJavaFile, fileName));
+    editor.waitActive();
+    editor.deleteAllContent();
+    editor.typeTextIntoEditor(text);
+    editor.waitActive();
+    editor.waitTextIntoEditor(text);
+    editor.waitTabFileWithSavedStatus(fileName);
+  }
+
+  private void changeTextFileForTest(String fileName, String text) {
+    projectExplorer.openItemByPath(String.format("%s/%s", PROJECT_NAME, fileName));
+    editor.waitActive();
```

Comment:
Please, divide operations with identical page objects, by empty rows, to improve readability.